### PR TITLE
Support form data parsing

### LINKED ISSUE #269: [fusion-plugin-rpc#3] Add support for FormData
This issue was migrated from [fusionjs/fusion-plugin-rpc#3](https://github.com/fusionjs/fusion-plugin-rpc/issues/3) and was originally reported by @lhorie.

---

Rationale: Currently, RPC always assume JSON data. Supporting FormData would enable the ability to do file uploads

### PR DESCRIPTION
**Pull request by joginder07**
Resolves #269

- Added formidable for parsing form data.
- Added test case.
- Added form-data as dev dependency to mock form data.

### CODE REVIEW

**Comment by joginder07** [comment]
@chrisdothtml , @ganemone , can you please review this.

File: fusion-plugin-rpc/src/server.js [hunk 1]
```diff
@@ -140,8 +142,30 @@ const pluginFactory: () => RPCPluginType = () =>
           const pathMatch = new RegExp(`${apiPath}([^/]+)`, 'i');
           const [, method] = ctx.path.match(pathMatch) || [];
           if (hasHandler(handlers, method)) {
+            let body;
             try {
-              await parseBody(ctx, () => Promise.resolve());
+              if (
+                ctx.req &&
```
Comment (ganemone) [inline fusion-plugin-rpc/src/server.js]: ctx.req should be always available

Comment (joginder07) [reply fusion-plugin-rpc/src/server.js]: yes ctx.req will be present but it is done to pass failing test cases. In some test cases which were already there req is missing.

**Comment by chrisdothtml** [comment]
!import

**Pull request closed without merging**